src/sagemaker/jumpstart/types.py

```diff
@@ -736,6 +736,334 @@ def _get_regional_property(
+        self.hosting_instance_type_variants: Optional[JumpStartInstanceTypeVariants] = (
+            JumpStartInstanceTypeVariants(json_obj["hosting_instance_type_variants"])
+            if json_obj.get("hosting_instance_type_variants")
+            else None
+        )
+        self.tags: Optional[List[str]] = json_obj.get("tags")
+        self.environment_variables = [
+            JumpStartEnvironmentVariable(env_variable)
+            for env_variable in json_obj.get("environment_variables", [])
+        ]
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartInferenceComponent object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+
+class JumpStartTrainingComponent(JumpStartDataHolderType):
+    """Data class JumpStart Training components."""
+
+    __slots__ = [
+        "component_name",
+        "tags",
+        "supported_training_instance_types",
+        "training_artifact_key",
+        "training_instance_type_variants",
+        "supported_incremental_training_configs",
+        "supported_inference_configs",
+    ]
+
+    def __init__(self, spec: Optional[Dict[str, Any]]):
+        """Initializes a JumpStartTrainingComponent object from its json representation.
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of training config component.
+        """
+        self.from_json(spec)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of training config components.
+        """
+        if json_obj is None:
+            return
+
+        self.component_name: str = json_obj["component_name"]
+        self.supported_training_instance_types: Optional[List[str]] = json_obj.get(
+            "supported_training_instance_types"
+        )
+        self.training_artifact_key: Optional[str] = json_obj.get("training_artifact_key")
+        self.training_instance_type_variants: Optional[JumpStartInstanceTypeVariants] = (
+            JumpStartInstanceTypeVariants(json_obj["training_instance_type_variants"])
+            if json_obj.get("training_instance_type_variants")
+            else None
+        )
+        self.tags: Optional[List[str]] = json_obj.get("tags")
+        self.supported_incremental_training_configs: Optional[List[str]] = json_obj.get(
+            "supported_incremental_training_configs"
+        )
+        self.supported_inference_configs: Optional[List[str]] = json_obj.get(
+            "supported_inference_configs"
+        )
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartTrainingComponent object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+
+class JumpStartAlternativeConfig(JumpStartDataHolderType):
```

Comment:
Can we rename `JumpStartDeploymentConfig`?

Reply:
imo `DeploymentConfig` is an overloaded term because "deployment" often means Hosting. I'd lobby for `JumpStartPresetConfig`